src/data/fin/basic.lean

```diff
@@ -1363,6 +1367,16 @@ instance (n : ℕ) [ne_zero n] : add_comm_group (fin n) :=
   ..fin.add_comm_monoid n,
   ..fin.has_neg n }
 
+/-- Note this is more general than `fin.add_comm_group` as it applies (vacuously) to `fin 0` too. -/
+instance (n : ℕ) : has_involutive_neg (fin n) :=
+{ neg := has_neg.neg,
+  neg_neg := nat.cases_on n fin_zero_elim (λ i, neg_neg) }
+
+/-- Note this is more general than `fin.add_comm_group` as it applies (vacuously) to `fin 0` too. -/
+instance (n : ℕ) : is_cancel_add (fin n) :=
+{ add_left_cancel := nat.cases_on n fin_zero_elim (λ i _ _ _, add_left_cancel),
+  add_right_cancel := nat.cases_on n fin_zero_elim (λ i _ _ _, add_right_cancel) }
```

Comment:
`add_left_cancel_semigroup`? Or do we migrate to `is_*` predicates?

Reply:
Otherwise LGTM
bors d+

Reply:
I ended up having all three, since none imply the others.